server/src/test/java/org/elasticsearch/indices/recovery/RecoveryTargetTests.java

```diff
@@ -93,12 +93,10 @@ public T serializeDeserialize() throws IOException {
         }
 
         protected T deserialize(StreamInput in) throws IOException {
-            T obj = createObj();
-            obj.readFrom(in);
-            return obj;
+            return createObj(in);
         }
 
-        abstract T createObj();
+        abstract T createObj(StreamInput in) throws IOException;
```

Comment:
We need to make T extends from Writable.

Reply:
Thanks @dnhatn , I pushed 0bd175770a16c1bdd23b67186d0daadfd7accaef